src/utils/transformer-utils.ts

```diff
@@ -1,188 +1,320 @@
+        _type: "linkMark",
         _key: guid,
         value: value,
         childCount: childCount
-    }
-}
-
-export const createComponentBlock = (guid: string, reference: IReference): IPortableTextComponent => {
-    return {
-        _type: 'component',
-        _key: guid,
-        component: reference
-    }
-}
-
-export const isBlock = (block: IPortableTextItem): block is IPortableTextParagraph =>
-    block._type === 'block';
-
-export const isSpan = (span: IPortableTextItem): span is IPortableTextSpan =>
-    span._type === 'span';
-
-export const compose = <T>(firstFunction: (argument: T) => T, ...functions: Array<(argument: T) => T>) =>
-    functions.reduce((previousFunction, nextFunction) => value => previousFunction(nextFunction(value)), firstFunction);
-
-export const findLastIndex = <T>(arr: T[], predicate: (value: T) => boolean): number => {
-    for (let i = arr.length - 1; i >= 0; i--) {
-        if (predicate(arr[i])) {
-            return i;
-        }
-    }
-    return -1;
-}
-
-export const resolveTable = (table: IPortableTextTable, resolver: (value: any) => string): string => {
-    let tableHtml = '<table><tbody>';
-    const resolveCell = (cell: IPortableTextTableCell) => {
-        tableHtml += '<td>';
-        for (let j = 0; j < cell.childBlocksCount; j++) {
-            tableHtml += resolver(cell.content);
-        }
-        tableHtml += '</td>';
     };
-    for (let i = 0; i < table.numColumns; i++) {
-        const currentRow = table.rows[i];
-        tableHtml += '<tr>';
-        currentRow.cells.forEach(resolveCell);
-        tableHtml += '</tr>';
-    }
-    tableHtml += '</tbody></table>';
-    return tableHtml;
-}
-
-export const getAllNewLineAndWhiteSpace = /\n\s*/g;
\ No newline at end of file
+};
+
+export const createComponentBlock = (
+  guid: string,
+  reference: Reference
+): PortableTextComponent => {
+  return {
+    _type: "component",
+    _key: guid,
+    component: reference,
+  };
+};
+
+export const compose = <T>(
+  firstFunction: (argument: T) => T,
+  ...functions: Array<(argument: T) => T>
+) =>
+  functions.reduce(
+    (previousFunction, nextFunction) => (value) =>
+      previousFunction(nextFunction(value)),
+    firstFunction
+  );
+
+/**
+ * Renders a portable text table to HTML or a Vue virtual DOM node.
+ * 
+ * @param {PortableTextTable} table - The portable text table object to render.
+ * @param {(value: PortableTextBlock[]) => string} resolver - A function that resolves 
+ *        the content of each cell in the table.
+ * @param {Function} [vueRenderFunction] - An optional Vue render function for generating 
+ *        virtual DOM nodes. If provided, the table is rendered as Vue nodes, otherwise 
+ *        as an HTML string.
+ * @returns {string | VueNode} The rendered table as an HTML string or Vue virtual DOM node.
+ */
+export const resolveTable = (
+  table: PortableTextTable,
+  resolver: (value: PortableTextBlock[]) => string,
+  vueRenderFunction?: Function
```

Comment:
I don't like the function plugged in here. Are you going to add parameters for all the frameworks out there? The function doesn't seem to be vue specific. If it is, I would make it clear, maybe separate it into a file with vue handlers. Otherwise, I would try to separate any framework-specific handling into separate functions and files to make the distinction clear and easy to adjust separately.

Reply:
split and resolved in https://github.com/kontent-ai/rich-text-resolver-js/pull/39/commits/5bcc38a09f412a4928b1ecc3f5e351b122d9b0ec